src/endpoints/auth/auth.controller.ts

```diff
@@ -0,0 +1,139 @@
+      required: ['email', 'password'],
+    },
+    response: {
+      200: signUpOutputSchema,
+    },
+  },
+};
+
+const validateEmailOptions = {
+  schema: {
+    tags: ['auth'],
+    summary: 'Verify email here',
+    body: {
+      type: 'object',
+      properties: {
+        code: { type: 'string', minLength: 6, maxLength: 6 },
+      },
+      required: ['code'],
+    },
+    headers: {
+      type: 'object',
+      properties: {
+        authorization: { type: 'string' },
+      },
+      required: ['authorization'],
+    },
+    response: {
+      200: valitdateEmailOutputSchema,
+    },
+  },
+};
+
+export const signUpRouter: FastifyPluginAsync<FastifyPluginOptions> = async (server, options) => {
+  server.post<{ Body: IBodySignUp }>('/sign-up', signUpOptions, async (req, reply) => {
+    const { email, password } = req.body;
+
+    // todo 28.11.2021: remove later - only here for testing
+    await UserEntity.delete({ email });
+
+    const user = await UserEntity.findOne({ email });
+
+    if (user) {
+      throw createError(400, 'User exists.');
+    }
+
+    const sessionKey = randomUUID();
+    const salt = randomUUID();
+    const passwordHash = HashingService.createHash(password, sessionKey);
+    const dataUser = UserEntity.create({ email, passwordHash });
+
+    const { accessToken, refreshToken, accessTokenExpiresIn, refreshTokenExpiresIn } = await JWTService.generateTokens({
+      sessionKey,
+      jwtSecret: secret,
+      accessDeathDate,
+      refreshDeathDate,
+    });
+
+    dataUser.sessionKey = sessionKey;
+    dataUser.salt = salt;
+    await dataUser.save();
+
+    const { code, expiresAt } = VerificationCodeService.createCode();
+
+    const dataCodes = VerificationCodesEntity.create({ userId: dataUser._id, code, expiresAt });
+    await dataCodes.save();
+
+    await EmailService.sendMessageToEmail(email, code);
+
+    return { accessToken, refreshToken, accessTokenExpiresIn, refreshTokenExpiresIn };
+  });
+};
+
+export const validateEmailRouter: FastifyPluginAsync<FastifyPluginOptions> = async (server, options) => {
+  server.post<{ Body: IBodyValidateEmail; Headers: IHeadersValidateEmail }>(
+    '/email/validate',
+    validateEmailOptions,
+    async (req, reply) => {
+      const { code } = req.body;
+      const { authorization } = req.headers;
+
+      const sessionKey = await JWTService.decodeToken(authorization, secret);
+
+      if (!sessionKey) {
+        throw createError(400, 'Invalid access token.');
+      }
+
+      const user = await UserEntity.findOne({ sessionKey });
+
+      if (!user) {
+        throw createError(401, 'Invalid access token.');
+      }
+
+      const verificationCode = await VerificationCodesEntity.findOne({ userId: user._id });
+
+      if (!verificationCode || verificationCode.code !== code) {
+        throw createError(401, 'Invalid code sent.');
+      }
+
+      const date = DateTime.utc();
+      const expiresAt = DateTime.fromJSDate(verificationCode.expiresAt);
```

Comment:
Rename to `codeExpiresAt`. If the handler was bigger, you would forget what the abstract `expiresAt` means.